lib/ical/duration.js

```diff
@@ -273,8 +273,20 @@ class Duration {
       let str = "";
       if (this.isNegative) str += "-";
       str += "P";
-      if (this.weeks) str += this.weeks + "W";
-      if (this.days) str += this.days + "D";
+      console.log(this.weeks)
+      if(this.weeks){
+        if (this.days || this.hours || this.minutes || this.seconds){
+          str += (this.weeks*7 + this.days + "D");
```

Comment:
Can you run the linter on your code? I see a few things I think it might complain on re spacing and brackets.

I think we should go with this:

```javascript
str += (this.weeks * 7 + this.days) + "D";
```

to make a the arithmetic addition a bit more clear